x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -1006,39 +1008,50 @@ public LogicalPlan apply(LogicalPlan p) {
         }
     }
 
+    private static class PercentileKey extends Tuple<Expression, PercentilesConfig> {
+        PercentileKey(Percentile per) {
```

Comment:
A small nit - to avoid having two constructors, it would be possible make the base class public so that percentilesConfig can be accessed. Since the rest of the methods are not, it keeps the encapsulation while removing redundancy.